data/basics/userBalance.go

```diff
@@ -379,6 +380,16 @@ type AssetParams struct {
 	Clawback Address `codec:"c"`
 }
 
+func (app AppIndex) ToBeHashed() (protocol.HashID, []byte) {
+	buf := make([]byte, 8)
+	binary.BigEndian.PutUint64(buf, uint64(app))
+	return protocol.AppIndex, buf
```

Comment:
this name `protocol.AppIndex` was confusing, did not get it is a prefix

Reply:
That's the way these are all written.  See program.go for example.  We could change them all, I guess.  The only think making this slightly odder is that I had to turn the uint64 into a []byte.